crates/abd-clam/src/core/graph/_graph.rs

```diff
@@ -760,9 +748,138 @@ impl<'a, U: Number> Graph<'a, U> {
     ///
     /// # Errors
     ///
-    /// If the specified cluster is not part of the graph, an error message is returned.
+    /// If the specified cluster is not part of the graph or if `with_eccentricities` was not called, an error message is returned.
     pub fn eccentricity(&'a self, c: &'a Cluster<U>) -> Result<usize, String> {
-        self.assert_contains(c)?;
-        Ok(self.unchecked_eccentricity(c))
+        self.frontier_sizes(c).map_or_else(
+            |_| Err("Please call with_eccentricities before using this method".to_string()),
+            |frontier_sizes| Ok(frontier_sizes.len()),
+        )
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
i added that test as well as checking if the distance between two clusters is 0 if their value in the adjacency matrix is 0. I also checked if the distance is not zero if there is an edge but this could possibly lead to a false positive in a rare occurence where two clusters have an edge of length zero. Not sure if I should keep the second check.